lib/src/local_working_copy.rs

```diff
@@ -1391,7 +1434,13 @@ impl FileSnapshotter<'_> {
             };
             let new_tree_values = match new_file_type {
                 FileType::Normal { executable } => self
-                    .write_path_to_store(repo_path, disk_path, &current_tree_values, executable)
+                    .write_path_to_store(
+                        repo_path,
+                        disk_path,
+                        &current_tree_values,
+                        maybe_current_file_state,
```

Comment:
nit: at your option, the state could be mapped here to `Option<MaterializedConflictData>`, but I have no idea which is better overall.